### PR TITLE
Adding a new naming scheme used by xenon

### LINKED ISSUE #4336: Xenon LFN2PFN function
Motivation
----------
New xenon lfn2pfn function.

### PR DESCRIPTION
**Pull request by lucascottolavina**
protocols : Adding a new naming scheme used by xenon
------------------

On behalf of XENON Collaboration, we propose to include in the Rucio code a new scheme that will be used to store XENONnT data on tape storages. The algorithm is deterministic and it is meant to minimize the number of subdirectories and files inside any directory, at any level.

### CODE REVIEW

File: lib/rucio/rse/protocols/protocol.py [hunk 1]
```diff
@@ -199,6 +199,24 @@ def __ligo(scope, name, rse, rse_attrs, protocol_attrs):
...
+        :param rse: RSE for PFN (ignored)
+        :param rse_attrs: RSE attributes for PFN (ignored)
+        :param protocol_attrs: RSE protocol attributes for PFN (ignored)
+        :returns: Path for use in the PFN generation.
+        """
+        del rse
+        del rse_attrs
+        del protocol_attrs
+
+        return '%s/%s/%s/%s' % (scope[0:7], scope[4:len(scope)], name.split('-')[0]+"-"+name.split('-')[1], name)
```
Comment (mlassnig) [inline lib/rucio/rse/protocols/protocol.py]: Hi, looks good to me. Please don't forget to add yourself to the author list at the top of the file.

Comment (lucascottolavina) [reply lib/rucio/rse/protocols/protocol.py]: Thanks, done

Comment (mlassnig) [reply lib/rucio/rse/protocols/protocol.py]: github actions seemed to have a hiccup, i've restarted the unittests

Comment (lucascottolavina) [reply lib/rucio/rse/protocols/protocol.py]: I restarted the tests again after fixing a couple of style warning (flake8 apparently is very meticolous). They were just a missing space before and after the sum operator and I removed a tab in an empty line. Now also the autotests with client_syntax option have been completed successfully. All checks have passed now.

**Review by mlassnig — DISMISSED** [review]

**Comment by bari12** [comment]
Looks good 👍 I will include this in `1.24.4`
I believe Xenon is still on `1.20 LTS` though. Would you prefer this also to be backported to `1.20 LTS` and `1.23 LTS`?

**Comment by bari12** [comment]
Please rebase this, there have been changes on that file.

**Comment by lucascottolavina** [comment]
> Looks good +1 I will include this in `1.24.4`
> I believe Xenon is still on `1.20 LTS` though. Would you prefer this also to be backported to `1.20 LTS` and `1.23 LTS`?

yes, please, we will continue using the 1.20 LTS and if it possible to have it also on 1.23 it would be useful in case we want to jump to the next LTS

**Comment by bari12** [comment]
Ok,
This will go then to 
`1.24.5`
`1.23.13` (LTS)
`1.20.13` (LTS)
For the LTS releases I only plan to do them beginning of March though.

**Review by bari12 — APPROVED** [review]

**Pull request merged**